code/frameworks/vue3-vite/src/plugins/vue-component-meta.ts

```diff
@@ -0,0 +1,224 @@
+                })
+                // remove unwanted duplicated "$slots" expose
+                .filter((expose) => {
+                  if (expose.name === '$slots') {
+                    const slotNames = meta.slots.map((slot) => slot.name);
+                    return !slotNames.every((slotName) => expose.type.includes(slotName));
+                  }
+                  return true;
+                });
+
+            return {
+              exportName,
+              displayName: exportName === 'default' ? getFilenameWithoutExtension(id) : exportName,
+              ...meta,
+              exposed,
+              sourceFiles: id,
+            };
+          });
+
+        // if there is no component meta, return undefined
+        if (metaSources.length === 0) return undefined;
+
+        const s = new MagicString(src);
+
+        metaSources.forEach((meta) => {
+          const isDefaultExport = meta.exportName === 'default';
+          const name = isDefaultExport ? '_sfc_main' : meta.exportName;
+
+          // we can only add the "__docgenInfo" to variables that are actually defined in the current file
+          // so e.g. re-exports like "export { default as MyComponent } from './MyComponent.vue'" must be ignored
+          // to prevent runtime errors
+          if (new RegExp(`export {.*${name}.*}`).test(src)) {
+            return;
+          }
+
+          if (!id.endsWith('.vue') && isDefaultExport) {
+            // we can not add the __docgenInfo if the component is default exported directly
+            // so we need to safe it to a variable instead and export default it instead
+            s.replace('export default defineComponent(', 'const _sfc_main = defineComponent(');
+            s.append('\nexport default _sfc_main;');
+          }
+
+          s.append(`\n;${name}.__docgenInfo = ${JSON.stringify(meta)}`);
+        });
+
+        return {
+          code: s.toString(),
+          map: s.generateMap({ hires: true, source: id }),
+        };
+      } catch (e) {
+        return undefined;
+      }
+    },
+  };
+}
+
+/**
+ * Creates the vue-component-meta checker to use for extracting component meta/docs.
+ */
+async function createChecker() {
+  const checkerOptions: MetaCheckerOptions = {
+    forceUseTs: true,
+    noDeclarations: true,
+    printer: { newLine: 1 },
+  };
+
+  const projectRoot = getProjectRoot();
+  const projectTsConfigPath = path.join(projectRoot, 'tsconfig.json');
+
+  if (await fileExists(projectTsConfigPath)) {
+    // TODO: tsconfig that uses references is currently not supported by vue-component-meta
+    // find a temp workaround for this
+    // see: https://github.com/vuejs/language-tools/issues/3896
+
+    // prefer the tsconfig.json file of the project to support alias resolution etc.
+    return createComponentMetaChecker(projectTsConfigPath, checkerOptions);
+  }
+
+  return createComponentMetaCheckerByJsonConfig(projectRoot, { include: ['**/*'] }, checkerOptions);
```

Comment:
you can pass extend this with the package tsconfig like this 
```
{
		"extends": "../tsconfig.json",
		"include": [
			"**/*",
		],
	},
	```

Reply:
you can use the standard one used in vue language tools https://github.com/vuejs/language-tools/blob/master/packages/component-meta/tsconfig.json

Reply:
> you can extend this with the package tsconfig like this

We can not use any arbitrary tsconfig, it has to be exactly the tsconfig of the project. See this example which is taken from a new Vue starter application:
```json
// tsconfig.json
{
  "files": [],
  "references": [
    { "path": "./tsconfig.node.json" }, // not relevant for this example
    { "path": "./tsconfig.app.json" },
    { "path": "./tsconfig.vitest.json" } // not relevant for this example
  ]
}
```

```json
// tsconfig.app.json
{
  "extends": "@vue/tsconfig/tsconfig.dom.json",
  "include": ["env.d.ts", "src/**/*"],
  "compilerOptions": {
    // ...
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```

And now you have a component in your project that uses `@/` for imports:
```vue
<script lang="ts" setup>
import type { MyProps } from "@/types";

const props = defineProps<MyProps>();
</script>
```

`vue-component-meta` can not resolve the type/docs correctly so they will not show up in Storybook. Therefore we should prefer the actual tsconfig of the project over the non-tsconfig checker.